code/game/machinery/newscaster.dm

```diff
@@ -1044,3 +1060,72 @@ GLOBAL_LIST_EMPTY(allCasters)
 			add_fingerprint(user)
 	else
 		return ..()
+
+
+/*
+
+	A small datum for horoscopes (possibly implimented into the newscaster or a pda app?)
+	-candycane/etherware
+
+*/
+
+/datum/horoscope
+
+	var/has_read  // lazylist
+
+	var/static/list/positive = list(
+		"You will find something of great value in maintenance.",
+		"You will affect the station's balance.",
+		"You will support the station greatly.",
+		"You will come across unexpected glory.",
+		"You will meet a pleasant companion.",
+		"You're about to hit it big."
+	)
+
+	var/static/list/negative = list(
+		"A superior will deceive you,",
+		"You will find great sorrow in maintenance.",
+		"You will be slipped.",
+		"The station will be irreparably injured.",
+		"Your life hangs in the balance, tipping towards tradgedy.",
+		"You will be betrayed by someone close to you."
+	)
+
+	/// if set, will always be returned by get_horoscope
+	var/admin_msg
+	/// the amount your admin mood will be worth
+	var/admin_mood_amount = 1
+
+/datum/horoscope/New()
+	RegisterSignal(SSdcs, COMSIG_NEW_DAY, .proc/reset_readers)
+	. = ..()
+
+/datum/horoscope/proc/check_horoscope(mob/living/carbon/human/reader)
+	if(!reader || (reader?.ckey in has_read))
+		return
+	return TRUE
+
+/datum/horoscope/proc/get_horoscope(mob/living/carbon/human/reader)
+	if(!reader)
+		return
+	if(reader?.ckey in has_read)
+		return has_read[reader.ckey]
+
+	if(admin_msg)
+		if(admin_mood_amount)
+			SEND_SIGNAL(reader, COMSIG_ADD_MOOD_EVENT, "horoscope_admin", admin_mood_amount > 0 ? /datum/mood_event/horo_good : /datum/mood_event/horo_bad)
+		log_game("[reader] has gotten the admin horoscope: [admin_msg]")
+		return admin_msg
+
+	if(prob(50))
+		SEND_SIGNAL(reader, COMSIG_ADD_MOOD_EVENT, "horoscope", /datum/mood_event/horo_good)
+		. = pick(positive)
+	else
+		SEND_SIGNAL(reader, COMSIG_ADD_MOOD_EVENT, "horoscope", /datum/mood_event/horo_bad)
+		. = pick(negative)
+
+	log_game("[reader] has gotten the horoscope: [.]")
+	has_read[reader.ckey] = .
```

Comment:
if you die and have a new body the same day, you get the same horoscope.

Change reader.ckey to `REF(reader)`